gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanJobLauncher.java

```diff
@@ -49,7 +58,7 @@
  *
  * @author Yinan Li
  */
-public class AzkabanJobLauncher extends AbstractJob {
+public class AzkabanJobLauncher extends AbstractJob implements ApplicationLauncher, JobLauncher {
```

Comment:
True, but I think its more like `CliLocalJobLauncher extends LocalJobLauncher` or `CliMRJobLauncher extends MRJobLauncher` - but I wanted to avoid using `extends` since many of these classes already `extend` some other class.

`JobLauncherFactory` only has two publicly available launch types, `LOCAL` and `MAPREDUCE` and provides implementations for both. I could add the `AzkabanJobLauncher` here, but it wouldn't provide any benefit.

I think `JobLauncherFactory` is useful for creating `JobLauncher`s in code, but that shouldn't prevent creation of `JobLauncher`s that are directly invoked using a `main` method.